### ISSUE TITLE
Separate Token Retrieval from User Update

### ISSUE THREAD

**Issue by JANA-itsaishvili** [auth0/auth0-react#787]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/auth0-react#readme), [Examples](https://github.com/auth0/auth0-react/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-react/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/auth0-react/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-react/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

I recently ran into an infinite loop scenario where I needed to get the token and pass it with the user object to another function inside useEffect. That function is memoized in useCallback, with only navigate (react-router-dom) dependency. The only variable element in the useEffect dependency array is the user object from useAuth0.

I looked at the code, and the reducer is setting the user object to a newly decoded one from getUser on each GET_ACCESS_TOKEN_COMPLETE action. This action is called every time with getAccessTokenSilenty, in the finally block. This means it's impossible to retrieve the access token alone, without regenerating the user object.

I know there are many ways to solve this, but I think there shold be an option or a new method that simply returns the access token without tempering with the user object. I'm guessing the reason behind doing it the current way is to keep the user object in sync with the decoded token claims. If not, it makes no sense to update the user object, considering how React works.

I simply need to get the token to make an API request. I shouldn't have to worry about the user object in the process.

### Describe the ideal solution

- add a new method, that only fetches the token, without updating the user context, like getAccessTokenOnly.
- update reducer logic to compare previous state with action payload, but this would break current user expectations
- memoize the user in the provider

Ideally, if there is a relationship between the access_token and the user, the solution will come from the vanilla js package.

### Alternatives and current workarounds

I am calling the js sdk manually in this case, inside useEffect. I actually rebuilt the react package, but the reducer logic and provider are the same, as far as this feature is concerned. I instantiate the client outside of the context, and let the provider closer over it.

**Comment by gyaneshgouraw** [comment]
Hi @JANA-itsaishvili , thanks for reporting this! We appreciate you bringing it to our attention — we will review it and follow up with an update shortly.

**Comment by yogeshchoudhary147** [comment]
## Issue Status: ✅ RESOLVED

This issue was **fixed in `@auth0/auth0-react` v2.0.0** when it upgraded to `@auth0/auth0-spa-js` v2.0.0, which included PR [#975](https://github.com/auth0/auth0-spa-js/pull/975).

### The Problem (Pre-v2.0.0)
`getAccessTokenSilently()` created a new user object on every call, causing infinite loops in `useEffect` hooks that depended on the `user` object.

### The Fix
`@auth0/auth0-spa-js` v2.0.0 implemented in-memory caching of the decoded ID token, returning the same object reference when token content hasn't changed. This fix was inherited by `@auth0/auth0-react` v2.0.0.

### Verification
Tested on `@auth0/auth0-react: ^2.11.0`:
- ✅ User object stable across multiple token fetches
- ✅ `useEffect` runs only once
- ✅ No infinite loops

### Upgrade
```bash
npm install @auth0/auth0-react@latest
```

**Required:** `@auth0/auth0-react` >= 2.0.0

---

**Closing** as resolved. If you're still experiencing issues on v2.0.0+, please open a new issue with reproduction details.
